.dotnet/api/OpenAI.netstandard2.0.cs

```diff
@@ -1612,6 +1612,8 @@ public class FileClient {
         public FileClient(ApiKeyCredential credential, OpenAIClientOptions options = null);
         protected internal FileClient(ClientPipeline pipeline, Uri endpoint, OpenAIClientOptions options);
         public virtual ClientPipeline Pipeline { get; }
+        public virtual IncrementalUpload CreateIncrementalUpload(BinaryContent content, RequestOptions options = null);
```

Comment:
What is the scenario for these incremental uploads? I am asking as I am not sure the users cares that they are incremental. For example, we have a very similar feature in storage where you can upload blocks and then the server stitches them into a single blob. But users rarely, or ever, care about the blocks or stitching. All they care about is that the upload of their file can be restarted if it fails after uploading some part of the file. And so the question: should our ApIs be similar to how storage APIs work? If yes, then I don't mind shipping low level protocol method for the low level stitching protocol, but we need to make sure we can evolve the subclient to look like the storage APIs and I dont think these can because this method takes BinaryContent which is the first part, isn't it?

Reply:
This is the community post that briefly introduced it:
https://community.openai.com/t/new-api-feature-multipart-uploads/870493

> You can upload a large file as a series of parts, upload parts in parallel to speed things up, and pause the upload in the middle if you get interrupted.

> Uploads expire after one hour from when they are initiated. The size of all files uploaded for a user can be up to 10 GB, and the size of all files uploaded for a project can be up to 100 GB.

It certainly seems like an advanced/niche use case, but there was presumably enough signal from customers that they wanted to be able to do this parallelization/pausing/etc. for them to implement it.

What can we do here to, in generality, ship protocol support very quickly and with very little architectural overhead while not causing undue problems for ourselves in the future? I'd like to ensure we don't make regrettable decisions, but I'd *also* like to ensure we can move much, much faster than we are with parity updates. Embracing self-criticism a bit, we've done an absolutely horrible job at keeping up with the spec; the feature was released 26 days ago, the first draft of this PR was out 25 days ago, and here we're still not releasing even basic protocol support for it -- further, since it's a new operation, there's no workaround of "just use the protocol method."

Can we, e.g., just go back to doing "dumb" full protocol support on the base client (no subclient) and then accepting that the convenience layer may not match protocol surface one-to-one? Is there any inherent reason that a convenience subclient construct couldn't rely on backref calls to a parent client?